source/common/config/grpc_mux_impl.cc

```diff
@@ -359,6 +390,19 @@ void GrpcMuxImpl::processDiscoveryResources(const std::vector<DecodedResourcePtr
       auto it = resource_ref_map.find(watched_resource_name);
       if (it != resource_ref_map.end()) {
         found_resources.emplace_back(it->second);
+      } else if (isXdstpWildcard(watched_resource_name)) {
+        // See if the resources match the xdstp wildcard entry.
+        // Note: the assumption is that we don't have singleton and collection subscriptions in the
+        // same watch. For that reason, we do not need to check for duplicates in adding to
+        // `found_resources`.
```

Comment:
nit: even if we there are a singleton and collection watchers, I think it is ok, because of the `else if` above.
Maybe remove or rephrase it as follows:
```suggestion
        // Note: although it is unlikely that Envoy will need to support a resource that is mapped to both a singleton and collection watch,
        // this code still supports this use case.
```